modules/openapi-generator/src/main/resources/rust/model.mustache

```diff
@@ -77,6 +77,14 @@ impl Default for {{classname}} {
         }{{/-first}}{{/mappedModels}}{{/oneOf}}{{#oneOf}}{{#-first}}Self::{{{.}}}(Box::default()){{/-first}}{{/oneOf}}
     }
 }
+{{#oneOf}}
+
+impl From<{{{.}}}> for {{classname}} {
+    fn from(model: {{{.}}}) -> Self {
+        Self::{{{.}}}(Box::new(model))
+    }
+}
+{{/oneOf}}
```

Comment:
Note that `{{#oneOf}} {{{.}}} {{/oneOf}}` isn't enough here. 
Even against current code - there is a case with `mappedModels`, but actual issue is that current implementation in master have several issues, including wrong imports.

You can take a look at https://github.com/OpenAPITools/openapi-generator/pull/17915
I think this feature should be introduced only after the fixes